### DISCUSSION TITLE
m.recreateAllViews() correct usage

### DISCUSSION THREAD

**Discussion by jhwz** [simolus3/drift#1732] [Q&A] [answered]
Hey all!

First off just wanted to say this is a pretty awesome project, it has been a pleasure to use.

I was writing some migrations for my database as I'm adding some more features to an small app I have, and noticed the           `m.recreateAllViews()` method. 

I couldn't find much content on this other than the function comment saying
```dart
/// Drop and recreate all views. You should call it on every upgrade
```

I imagine no-one has gotten to writing documentation for this yet but I was wondering what the purpose of this is, and when it should be called? I'm currently assuming I can just call it at the end of my migration..?

**Comment by simolus3** [comment, marked as answer]
Thanks for starting this discussion! It's true that the documentation could be better here. From my knowledge, re-creating views is only really necessary when the definition of a view changes. With Dart-defined views, this can sometimes be the case without touching the view class at all (for instance because a referenced column from a table is renamed).

Dropping and re-creating views at the end of a migration is a fairly straightforward way to solve this, but other approaches are certainly possible too. I'll put this information on the website and make the dart docs of that method clearer.

**Reply by jhwz** [reply]
Sweet! Makes sense
